lib/spack/spack/ci.py

```diff
@@ -406,15 +427,27 @@ def find_matching_config(spec, ci_mappings):
     return None
 
 
-def release_jobs(parser, args):
-    env = ev.get_env(args, 'release-jobs', required=True)
+def pkg_name_from_spec_label(spec_label):
+    return spec_label[:spec_label.index('/')]
+
+
+def generate_gitlab_ci_yaml(env, print_summary, output_file):
```

Comment:
I think this method could be split up and pieces moved to core modules (e.g. in a `spack.ci` module) but don't do it now.  Overall this refactor improves a lot already and we should get it in first.

Reply:
Yep, I agree we should do that, but save it for a subsequent PR.